ckanext/datagovuk/plugin.py

```diff
@@ -273,8 +277,37 @@ def make_middleware(self, app, config):
             sentry_sdk.init(integrations=[FlaskIntegration()])
             return app
 
+    def before_create_or_update(self, context, resource):
+        """before_create_or_update - our own function. NOT a CKAN hook.
+        Contains shared code performed regardless of whether we are
+        creating or updating.
+        """
+
+        # If resource is an API, don't do anything special
+        if resource.get("format") == "API":
+            return
+        elif "upload" not in resource or resource["upload"] == "":
```

Comment:
```suggestion
        elif not resource.get("upload"):
```

Because `None` and `""` are falsey.

Reply:
I'd tried this before and the tests failed — turns out it's because a FieldStorage can also be falsey, which is the fault of my test setup.